src/DataCollectors/Microsoft.TestPlatform.Extensions.EventLogCollector/EventLogDataCollector.cs

```diff
@@ -0,0 +1,662 @@
+                }
+            }
+            catch (Exception ex)
+            {
+                EqtTrace.Warning(
+                    "Error occurred while trying to delete the temporary event log directory {0} :{1}",
+                    dirPath,
+                    ex);
+            }
+        }
+
+        private void StartCollectionForContext(DataCollectionContext dataCollectionContext, bool isSessionContext)
+        {
+            EventLogCollectorContextData eventLogContext;
+            lock (this.contextData)
+            {
+                if (this.contextData.TryGetValue(dataCollectionContext, out eventLogContext))
+                {
+                    if (EqtTrace.IsVerboseEnabled)
+                    {
+                        EqtTrace.Verbose(string.Format(
+                            CultureInfo.InvariantCulture,
+                            "EventLogDataCollector: Context data already in dictionary"));
+                    }
+                }
+                else
+                {
+                    eventLogContext =
+                        new EventLogCollectorContextData(isSessionContext ? int.MaxValue : this.maxEntries);
+                    this.contextData.Add(dataCollectionContext, eventLogContext);
+                }
+            }
+
+            foreach (string eventLogName in this.eventLogNames)
+            {
+                try
+                {
+                    // Create an EventLog object and add it to the eventLogContext if one does not already exist
+                    if (!this.eventLogContainers.ContainsKey(eventLogName))
+                    {
+                        var eventLogContainer = this.CreateEventLogContainer(eventLogName, eventLogContext);
+                        this.eventLogContainers.Add(eventLogName, eventLogContainer);
+
+                        if (EqtTrace.IsVerboseEnabled)
+                        {
+                            EqtTrace.Verbose(
+                                string.Format(
+                                    CultureInfo.InvariantCulture,
+                                    "EventLogDataCollector: Enabling collection of '{0}' events for data collection context '{1}'",
+                                    eventLogName,
+                                    dataCollectionContext.ToString()));
+                        }
+                    }
+                }
+                catch (Exception ex)
+                {
+                    this.logger.LogError(
+                        dataCollectionContext,
+                        new EventLogCollectorException(string.Format(CultureInfo.InvariantCulture, Resource.Execution_Agent_DataCollectors_EventLog_ReadError, eventLogName, Environment.MachineName), ex));
+                }
+            }
+        }
+
+        private void WriteCollectedEventLogEntries(
+            DataCollectionContext dataCollectionContext,
+            bool terminateCollectionForContext,
```

Comment:
When is `terminateCollectionForContext`:`fasle`?

Reply:
Keeping this field as is for now. 

This would be useful, in case more events are supported by DataCollection Framework.